### PR TITLE
Standardize Respository field

### PR DESCRIPTION
**Pull request by tc80**
- remove path attribute from legacy package'

All current `repository` attributes:

```
Summary of Keys in Object
repository: KEYS (3897): type
repository: KEYS (3897): url
repository: KEYS (4): directory
repository: [a/airbrake-browser.json a/angular.json m/material-ui.json r/react-is.json]

repository: KEYS (1): path
repository: [o/openajax-hub.json]

repository: KEYS (1): docs
repository: [p/photoset-grid.json]
```

### CODE REVIEW

**Comment by MattIPv4** [comment]
Do we currently support `directory` when cloning? If not, do we need to remove that and update any auto-update basePaths accordingly?

**Comment by tc80** [comment]
> Do we currently support `directory` when cloning? If not, do we need to remove that and update any auto-update basePaths accordingly?

Nope, we completely ignoring `directory`. Will investigate the packages... they may not even contain `directory` in their base paths anymore ...

```
// Repository represents a repository.
type Repository struct {
	Repotype string `json:"type"`
	URL      string `json:"url"`
}
```

**Comment by tc80** [comment]
@MattIPv4 I don't think we need to update any `basePath` fields because all four of the packages containing `directory` auto-update from npm. The `directory` is just useful to show where the code lives in their respective git repo...

I guess we can leave it alone -- what do you think?

**Comment by tc80** [comment]
Missing repository or not equal to "git":

Should `repository` be required? Will investigate the following:
```
[WebRupee](https://github.com/cdnjs/packages/blob/master/packages/W/WebRupee.json)
[amstockchart](https://github.com/cdnjs/packages/blob/master/packages/a/amstockchart.json)
[boexfi](https://github.com/cdnjs/packages/blob/master/packages/b/boexfi.json)
[canvasjs](https://github.com/cdnjs/packages/blob/master/packages/c/canvasjs.json)
[casualjs](https://github.com/cdnjs/packages/blob/master/packages/c/casualjs.json)
[chrome-frame](https://github.com/cdnjs/packages/blob/master/packages/c/chrome-frame.json)
[datejs](https://github.com/cdnjs/packages/blob/master/packages/d/datejs.json)
[dd_belatedpng](https://github.com/cdnjs/packages/blob/master/packages/d/dd_belatedpng.json)
[diff_match_patch](https://github.com/cdnjs/packages/blob/master/packages/d/diff_match_patch.json)
[ext-core](https://github.com/cdnjs/packages/blob/master/packages/e/ext-core.json)
[extjs](https://github.com/cdnjs/packages/blob/master/packages/e/extjs.json)
[fatcow-icons](https://github.com/cdnjs/packages/blob/master/packages/f/fatcow-icons.json)
[flipCounter](https://github.com/cdnjs/packages/blob/master/packages/f/flipCounter.json)
[galleriffic](https://github.com/cdnjs/packages/blob/master/packages/g/galleriffic.json)
[geo-location-javascript](https://github.com/cdnjs/packages/blob/master/packages/g/geo-location-javascript.json)
[jinplace](https://github.com/cdnjs/packages/blob/master/packages/j/jinplace.json)
[jqplugin](https://github.com/cdnjs/packages/blob/master/packages/j/jqplugin.json)
[jquery-ui-map](https://github.com/cdnjs/packages/blob/master/packages/j/jquery-ui-map.json)
[jquery.socialshareprivacy](https://github.com/cdnjs/packages/blob/master/packages/j/jquery.socialshareprivacy.json)
[js-url](https://github.com/cdnjs/packages/blob/master/packages/j/js-url.json)
[log4javascript](https://github.com/cdnjs/packages/blob/master/packages/l/log4javascript.json)
[meyer-reset](https://github.com/cdnjs/packages/blob/master/packages/m/meyer-reset.json)
[ninjaui](https://github.com/cdnjs/packages/blob/master/packages/n/ninjaui.json)
[openajax-hub](https://github.com/cdnjs/packages/blob/master/packages/o/openajax-hub.json)
[piecon](https://github.com/cdnjs/packages/blob/master/packages/p/piecon.json)
[screenfull.js](https://github.com/cdnjs/packages/blob/master/packages/s/screenfull.js.json)
[swfobject](https://github.com/cdnjs/packages/blob/master/packages/s/swfobject.json)
[tablefilter](https://github.com/cdnjs/packages/blob/master/packages/t/tablefilter.json)
[tinycon](https://github.com/cdnjs/packages/blob/master/packages/t/tinycon.json)
[toxiclibsjs](https://github.com/cdnjs/packages/blob/master/packages/t/toxiclibsjs.json)
[twilio.js](https://github.com/cdnjs/packages/blob/master/packages/t/twilio.js.json)
```

**Comment by MattIPv4** [comment]
👋 Haven't looked through that list of packages yet, but I think it'll probably be best to go with the following (pending review of the above list):

Require `repository`
Require `repository.type`, any value is fine
Require `repository.url`, any value is fine
Expect no other properties

**Comment by tc80** [comment]
Removed `directory`. We still need to fix the packages in the prior comment.

**Comment by MattIPv4** [comment]
For visibility, after some more discussion, we've decided on the following for now:

`repository` will be optional.

If `repository` is present, `type` & `url` will be expect, with no other keys being permitted.
`type` must match `/^git|hg|svn$/`

**Comment by tc80** [comment]
Fixed all repo types to match `^git|hg|svn$`

File: packages/t/toxiclibsjs.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "processing"
   ],
   "repository": {
-    "type": "plain file",
+    "type": "git",
     "url": "https://raw.github.com/hapticdata/toxiclibsjs/master/build/toxiclibs.min.js"
```
Comment (MattIPv4) [inline packages/t/toxiclibsjs.json:12]:
```suggestion
    "url": "https://github.com/hapticdata/toxiclibsjs"
```

**Comment by klausenbusk** [comment]
> `repository` will be optional.

Should `checker` be updated to reflect that?

https://github.com/cdnjs/tools/blob/bbbbc1287ea78c410c64ade7695172ce1337db51/cmd/checker/main.go#L411-L413

**Comment by MattIPv4** [comment]
https://github.com/cdnjs/tools/pull/137 will cover that

**Pull request merged**